File: test/qt-test.lua
--require 'qtcore'
--require 'qtgui'
--require 'qtuitools'
require 'qtuiloader'

app = QApplication.new(select('#',...) + 1, {'lua', ...})

--loader = QUiLoader.new()
--file = QFile.new('test.ui')
--win = QMainWindow.new()
--loader:load(file,win)
win = qtuiloader.load('test.ui')

horizontalSlider = win:findChild('horizontalSlider')
horizontalSlider:connect('2valueChanged(int)',function(self)
    print(self:value())
end)

win:show()
app.exec()

File: test/qt-test.js
var qt = require('qt');
var repl = require('repl');

app = new qt.QApplication;
timer = setInterval(app.processEvents, 0);

ui = new qt.QUiLoader();
win = ui.load(__dirname + '/test.ui');
win.show();
//r = qt.QMessageBox.warning('A','B',qt.STANDARDBUTTON.YES,qt.STANDARDBUTTON.NO,qt.STANDARDBUTTON.YES);
//console.log(r);

horizontalSlider = win.findChild('horizontalSlider');
horizontalSlider.off('valueChanged');
horizontalSlider.on('valueChanged',function(){console.log(this.value());})

repl.start('> ');
//clearInterval(timer);